src/DynamoPublish/ViewModels/PublishViewModel.cs

```diff
@@ -160,16 +159,29 @@ internal PublishViewModel(PublishModel model)
         private void OnPublish(object obj)
         {
             if (!model.IsLoggedIn)
+            {
                 model.Authenticate();
+            }
 
             if (!model.IsLoggedIn)
+            {
                 return;
+            }
 
-            var workspaceProperties = new WorkspaceProperties();
-            workspaceProperties.Name = Name;
-            workspaceProperties.Description = Description;
+            var workspaceProperties = new WorkspaceProperties
+            {
+                Name = Name,
+                Description = Description
+            };
+
+            var workspace = CurrentWorkspaceModel as HomeWorkspaceModel;
+
+            if (workspace == null)
+            {
+                throw new InvalidOperationException("The CurrentWorkspaceModel must be of type " + typeof(HomeWorkspaceModel).Name);
```

Comment:
The `PublishExtension` prevents this from ever occurring in the UI, but we throw an `Exception` for other potential use cases.